Source/SuperLinq/ToList.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace SuperLinq;
+
+public static partial class SuperEnumerable
+{
+	/// <summary>
+	/// Creates a <see cref="List{T}" /> from <see cref="IEnumerable{T}"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of list elements.</typeparam>
+	/// <param name="source">The <see cref="IEnumerable{T}"/> to create <see cref="List{T}" /> from.</param >
+	/// <param name="length">Expected length of source.</param>
+	/// <returns>
+	/// A <see cref="List{T}"/> that contains elements from the input sequence
+	/// </returns>
+	/// <exception cref="ArgumentNullException"><paramref name="source"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentOutOfRangeException"><paramref name="length"/> is <see langword="null"/>.</exception>
+	public static List<T> ToList<T>(this IEnumerable<T> source, int length)
+	{
+		Guard.IsNotNull(source);
+		Guard.IsGreaterThanOrEqualTo(length, 0);
+
+		var resultList = new List<T>(length);
+		resultList.AddRange(source);
```

Comment:
This is as or less efficient than using ToList (which has lots of optimizations, some based on the internal inferface [`IIListProvider`](https://source.dot.net/#System.Linq/System/Linq/IIListProvider.cs,780c77a1ae5f3e85,references)) in virtually all scenarios. The only exception I can think of would be if you somehow knew in advance the count of an enumerable that does not implement `ICollection` or `IIListProvider`, which is extremely rare.

If you still think this is worthwhile to add, a better implementation would be

````csharp
var resultList = new List<T>(source);
resultList.EnsureCapacity(int capacity);
return resultList;

Reply:
A lot of this applies to `ToArray` as well, especially since there are internal-only methods that allow copying collections into arrays.

Reply:
What about cases when I want to convert to list results of `Where`? Or some custom generator methods that use `yield return`?